discovery-infra/day2.py

```diff
@@ -290,3 +304,20 @@ def set_cluster_proxy(client, cluster_id, copy_proxy_from_cluster, args):
     https_proxy = args.https_proxy if args.https_proxy else copy_proxy_from_cluster.https_proxy
     no_proxy = args.no_proxy if args.no_proxy else copy_proxy_from_cluster.no_proxy
     client.set_cluster_proxy(cluster_id, http_proxy, https_proxy, no_proxy)
+
+def set_nodes_hostnames_if_needed(client,
+                                  terraform_cluster_dir_prefix,
+                                  namespace,
+                                  with_static_network_config,
+                                  has_ipv_4,
+                                  network_name,
+                                  cluster_id):
+    tf_folder = utils.get_tf_folder(terraform_cluster_dir_prefix, namespace)
+    tf = terraform_utils.TerraformUtils(working_dir=tf_folder)
+    libvirt_nodes = utils.extract_nodes_from_tf_state(tf.get_state(), network_name, consts.NodeRoles.WORKER)
+    if not has_ipv_4 or with_static_network_config:
```

Comment:
Why do you need to set hostnames with static network config even when you have IPv4?

Reply:
with static ips set, hostname is not set, since those ips are not configured in the terraform. This is also what will probably happen in production